fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/PullUpPredicateRule.java

```diff
@@ -0,0 +1,188 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+package com.starrocks.sql.optimizer.rule.tree;
+
+import com.starrocks.sql.optimizer.OptExpression;
+import com.starrocks.sql.optimizer.OptExpressionVisitor;
+import com.starrocks.sql.optimizer.Utils;
+import com.starrocks.sql.optimizer.base.ColumnRefSet;
+import com.starrocks.sql.optimizer.operator.logical.LogicalFilterOperator;
+import com.starrocks.sql.optimizer.operator.logical.LogicalJoinOperator;
+import com.starrocks.sql.optimizer.operator.scalar.BinaryPredicateOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ColumnRefOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ScalarOperator;
+import com.starrocks.sql.optimizer.task.TaskContext;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+public class PullUpPredicateRule implements TreeRewriteRule {
+
+    @Override
+    public OptExpression rewrite(OptExpression root, TaskContext taskContext) {
+        RewriteContext rewriteContext = new RewriteContext();
+        root.getOp().accept(new Visitor(), root, rewriteContext);
+
+        //Convert all remaining predicates to LogicalFilter
+        convertPredicateToLogicalFilter(root, rewriteContext);
+        return root;
+    }
+
+    private static class Visitor extends OptExpressionVisitor<Void, RewriteContext> {
+
+        public Visitor() {
+        }
+
+        @Override
+        public Void visit(OptExpression optExpression, RewriteContext context) {
```

Comment:
Should consider union, cte node and other node, union/cte maybe can‘t  pull up direct

Reply:
Project node
like:
```
select x0.v1, x, sum(v3) from (select v1, v2, v3, abs(1) x from t0) x0 group by x0.v1, x
```
It doesn't seem to support

Reply:
done